evopy/individual.py

```diff
@@ -1,33 +1,40 @@
 """Module containing the individuals of the evolutionary strategy algorithm."""
 import numpy as np
 
+from evopy.strategy import Strategy
+
 
 class Individual:
     """The individual of the evolutionary strategy algorithm.
 
     This class handles the reproduction of the individual, using both the genotype and the specified
     strategy.
     """
+    _EPSILON = 0.01
 
-    def __init__(self, genotype, strategy):
+    def __init__(self, genotype, strategy, strategy_parameters):
         """Initialize the Individual.
 
         :param genotype: the genotype of the individual
-        :param strategy: the strategy parameters of the individual
+        :param strategy: the strategy chosen to reproduce. See the Strategy enum for more
+                         information
+        :param strategy_parameters: the parameters required for the given strategy, as a list
         """
         self.genotype = genotype
-        self.strategy = strategy
+        self.length = len(genotype)
         self.fitness = None
-
-    def reproduce(self):
-        """Create a single offspring individual from the set genotype and strategy.
-
-        :return: an individual which is the offspring of the current instance
-        """
-        new_genotype = self.genotype + self.strategy * np.random.randn(len(self.genotype))
-        scale_factor = np.random.randn() / (2 * len(self.genotype))
-        new_strategy = max(self.strategy * np.exp(scale_factor), 0.01)
-        return Individual(new_genotype, new_strategy)
+        if strategy == Strategy.SINGLE_VARIANCE and len(strategy_parameters) == 1:
+            self.strategy = strategy
```

Comment:
`self.strategy = strategy` and `self.strategy_parameters = strategy_parameters` could be moved out of this if statement as they are present in all cases